ludwig/datasets/__init__.py

```diff
@@ -80,6 +63,71 @@ def get_dataset(dataset_name, cache_dir=None) -> Any:
     return loader_cls(config)
 
 
+@DeveloperAPI
+def load_dataset_uris(
+    dataset: Optional[Union[str, DataFrame]],
+    training_set: Optional[Union[str, DataFrame]],
+    validation_set: Optional[Union[str, DataFrame]],
+    test_set: Optional[Union[str, DataFrame]],
+    backend: Backend,
+) -> Tuple[
+    Optional[CacheableDataframe],
+    Optional[CacheableDataframe],
+    Optional[CacheableDataframe],
+    Optional[CacheableDataframe],
+]:
+    """Loads and returns any Ludwig dataset URIs as CacheableDataframes.
+
+    Returns the input unmodified for any non-Ludwig datasets.
+    """
+    dataset_out = dataset
+    training_set_out = training_set
+    validation_set_out = validation_set
+    test_set_out = test_set
+
+    # Check that any of the datasets begin with the `ludwig://` prefix denoting a Ludwig dataset URI
+    if dataset is not None:
+        if isinstance(dataset, str) and dataset.startswith(URI_PREFIX):
+            dataset_out = _load_cacheable_dataset(dataset, backend)
+    elif training_set is not None:
+        train_df = test_df = val_df = None
+        training_set_checksum = None
+        if isinstance(training_set, str) and training_set.startswith(URI_PREFIX):
+            # For the training set, we only want to use the TRAINING split of the dataset
+            dataset_name = training_set[len(URI_PREFIX) :]
+            loader = get_dataset(dataset_name)
+            train_df, test_df, val_df = loader.load(split=True)
+            training_set_checksum = str(loader.get_mtime())
+            train_df = backend.df_engine.from_pandas(train_df)
+            training_set_out = CacheableDataframe(df=train_df, name=training_set, checksum=training_set_checksum)
+
+        if isinstance(validation_set, str) and validation_set.startswith(URI_PREFIX):
+            if validation_set == training_set:
+                # Reuse the loaded DF from the training split
+                val_df = backend.df_engine.from_pandas(val_df)
```

Comment:
I might be missing something here, but won't `val_df` be None since we set it to that on line 93?

Reply:
Oh nvm I see it gets set in line 99